### PR TITLE
chore: Add action to publish jvm/android packages and move to maintained rust toolchain

### PR DESCRIPTION
**Pull request by augustocdias**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Adds a step in the CI to publish JVM and Android packages.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .github/workflows/publish-jvm.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          brew tap messense/macos-cross-toolchains
+          brew install x86_64-unknown-linux-gnu
+          cd ./core-crypto/crypto-ffi
+          cargo make copy-jvm-resources
+          cargo make copy-android-resources
+          cd ../kotlin
+          ./gradlew android:publishMavenPublicationToWireRepository
+          ./gradlew jvm:publishMavenPublicationToWireRepository
+          cd ../../wire-maven
+          git commit -am "Release of core-crypto $GITHUB_REF_NAME"
```
Comment (typfel) [inline .github/workflows/publish-jvm.yml:55]:
```suggestion
          git commit -am "chore: release of core-crypto $GITHUB_REF_NAME"
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -19,10 +19,8 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v3
 
-    - uses: actions-rs/toolchain@v1
+    - uses: dtolnay/rust-toolchain@stable
       with:
-        profile: minimal
-        toolchain: stable
         target: wasm32-unknown-unknown
```
Comment (OtaK) [inline .github/workflows/docs.yml:24]:
```suggestion
        targets: wasm32-unknown-unknown
```

**Review by OtaK — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -9,98 +9,61 @@ jobs:
...
         with:
-          profile: minimal
-          toolchain: stable
           components: rustfmt
       - uses: Swatinem/rust-cache@v1
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo fmt --all -- --check
```
[the remark this answers is not in the document]
Comment (augustocdias) [reply .github/workflows/rust.yml:16]: Unmaintained project

**Review by OtaK — CHANGES_REQUESTED** [review]
I guess it's just missing a rebase + squash now

**Comment by augustocdias** [comment]
> I guess it's just missing a rebase + squash now

We're changing it to publish to github packages instead of a separate repo

**Review by typfel — APPROVED** [review]

**Comment by OtaK** [comment]
> > I guess it's just missing a rebase + squash now
> 
> We're changing it to publish to github packages instead of a separate repo

I don't get it. Squash was needed to avoid this: 
image

**Comment by augustocdias** [comment]
> > > I guess it's just missing a rebase + squash now
> > 
> > 
> > We're changing it to publish to github packages instead of a separate repo
> 
> I don't get it. Squash was needed to avoid this: image
> 
> Also the fact that none of the commits respect conventional commits

It was squashed on merge. Why does it matter what is in this branch (which I deleted by the way). There's only one commit on develop. 

Also we're publishing a package now. If you check the side bar on the right of our repo you'll notice this now:

image

**Pull request merged**